### PR TITLE
Added support for Jinja2 3.1.2 and scikit-learn 1.0.2

### PR DESCRIPTION
**Pull request by dvorst**
When using Jinja2 3.1.2 and scikit-learn 1.0.2, the following errors occurred:
- AttributeError: module 'jinja2.ext' has no attribute 'with_'
- ModuleNotFoundError: No module named 'sklearn.metrics.scorer'
- ModuleNotFoundError: No module named 'sklearn.feature_selection.base'

The changes in 'html.py', 'permutation_importance.py' and 'transform.py' are such that these errors are resolved while maintaining support for older versions of jinja2 and scikit-learn.

'test_import.py' is used to test importing eli5 for both newer and older dependency versions.

Also see: https://github.com/TeamHG-Memex/eli5/issues/417

### CODE REVIEW

File: tests/test_import.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import jinja2
+import sklearn
+
+
+def test_import():
+	from packaging import version
+	if version.parse(jinja2.__version__) <= version.parse('3.0.3'):
+		raise ImportError('module "Jinja2" must be of a later version than "3.0.3".')
+	if version.parse(sklearn.__version__) < version.parse('1.0.2'):
+		raise ImportError('module "Jinja2" must be version "1.0.2" or later.')
```
Comment (lopuhin) [inline tests/test_import.py:10]: What is the goal of these checks? From the code above it seems that we do support older versions of these libraries? If we want to work only with versions specified here, it's better to specify this in setup.py.

**Review by lopuhin — COMMENTED** [review]
Hi @dvorst , thanks for the fixes 👍 I didn't run the tests but changes look good to me, left one comment below. However, would you mind submitting the PR against https://github.com/eli5-org/eli5/, if you don't mind?

File: tests/test_import.py (hunk 1 above)
Comment (dvorst) [reply tests/test_import.py:10]: Yeah, the test file is probably not really needed, I omitted it and also slightly changed some other stuff in the PR for https://github.com/eli5-org/eli5/ (the sklearn issue for example was already resolved)

**Comment by Hellisotherpeople** [comment]
Can we get a merge ASAP on this? ELI5 is now broken for most use-cases since you know, most people are running this on sklearn models... 

Also breaks my HF space: https://huggingface.co/spaces/Hellisotherpeople/Interpretable_Text_Classification_And_Clustering

**Comment by lopuhin** [comment]
@Hellisotherpeople this was merged in https://github.com/eli5-org/eli5/pull/19 and released in 0.12, thanks for a reminder
